pkg/asset/agent/image/agentpxefiles.go

```diff
@@ -118,6 +131,51 @@ func (a *AgentPXEFiles) PersistToFile(directory string) error {
 		return err
 	}
 
+	tmpdir, err := os.MkdirTemp("", pxeAssetsPath)
```

Comment:
A couple of points about this block:
* For the sake of clarity, it'd be recommended to extract the code specific to the kernel args into its own function
* Most of the current logic should be moved into the `Generate` method (at least up to the extraction of the relevant data, in this case I'd to the unmarshaling point). The `PersistToFile` method should contain only the logic relevant for writing the current asset data to the disk